GUIDELINES.md

```diff
@@ -171,6 +171,32 @@ Example:
 }
 ```
 
+### Request Nullable/Optional properties
+
+#### When to use `null` in properties
+
+1. When you want to update a property and unassign its value.
+2. If the property is optional and you are creating it you might want to use `null` to avoid the `if`s burden.
+
+#### When not to use `null` in properties
+
+1. Whenever you are updating a property that is optional/nullable and has a value assigned you don't want to remove.
+
+#### Nullable vs Optional
+
+When you set a property to `null` your intention is to remove any assigned value to it.
+When a property is optional you don't need to send the key in your request. The value will remain unchanged.
+
+**Note:** `Nullable` applies to the value of the property and `Optional` applies to the key of that property in the payload.
+
+#### Note on Empty strings
+
+Accepting empty strings does not have any meaning. If the goal is to unassign the value of a property then you should use `null` instead of an empty string.
```

Comment:
Q: do we need to specify this? We normally validate against this, right? (`null` or a non-empty string)

Reply:
I'm not sure that we do always validate against this. But lately, there has been some discussion regarding using empty strings in the values.

Reply:
The goal though is to make validation less strict, i.e. to accept empty strings in addition to `null`. This was discussed with @stivni in the context of the IBAN validation in the API. I will check to write a blog post on this.

Reply:
But do we handle empty strings as a `null` value then or as a value containing an empty string?

Reply:
(I vote for `null`)

Reply:
In most cases an empty string and `null` can be perceived as the same value, so if people send and empty string, but we don't want to store it like that, we should convert it first to `null` instead of not accepting it

Reply:
Can this decision be documented somewhere? (blogpost + meta repository f.e.?)